app/common/src/services/Backend.ts

```diff
@@ -937,8 +940,14 @@ export interface Asset<Type extends AssetType = AssetType> {
   readonly permissions: readonly AssetPermission[] | null
   readonly labels?: readonly LabelName[] | undefined
   readonly description?: string | undefined
+  /** Asset data for a project */
   readonly projectState: Type extends AssetType.project ? ProjectStateType : null
+  /** Asset data for a file */
   readonly extension: Type extends AssetType.file ? string : null
+  /** Asset data for a credential (secret) */
+  readonly serviceName?: Type extends AssetType.secret ? string : undefined
+  readonly expirationDate?: Type extends AssetType.secret ? dateTime.Rfc3339DateTime : undefined
+  readonly state?: Type extends AssetType.secret ? CredentialSecretState : undefined
```

Comment:
I don't like that these fields are flattened as they only are relevant for secrets (and not even all, only credentials), plus there's invariant - either all or none of them should be set - that this structure cannot enforce.

This requires modifying the `view::Asset` structure in the Rust lambdas. I will do the modification as part of https://github.com/enso-org/cloud-v2/pull/1821 
This is the right time to do this modification as we don't yet have usages of this (apart of #12084 which I will update before merging too), so better now then once we actually start depending on it.

Reply:
I will try to do the update tomorrow.

Reply:
Done in 5f847b1f9ff59c3b3f54ede1f5819b6e46fbe947

Reply:
Relevant backend change: https://github.com/enso-org/cloud-v2/pull/1821/commits/56853765dea9ebfcd7fd62e92e61663242a97334